test/integration/examples/apiserver_test.go

```diff
@@ -373,6 +375,112 @@ func waitForWardleRunning(ctx context.Context, t *testing.T, wardleToKASKubeConf
 	return directWardleClientConfig, nil
 }
 
+func TestAggregatedAPIServerRejectRedirectResponse(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	t.Cleanup(cancel)
+
+	backendServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
```

Comment:
can we add something like this in this handler:

```go
if strings.HasSuffix(r.URL.Path, "redirectTarget") {
   t.Error("backend called unexpectedly")
}
```
and redirect to `redirectedURL + "/redirectTarget"` so that if any layer follows the redirect it registers as an error

Reply:
Done